homeassistant/helpers/state.py

```diff
@@ -209,11 +210,20 @@ def state_as_number(state):
 
     Raises ValueError if this is not possible.
     """
-    if state.state in (STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON,
-                       STATE_OPEN, STATE_HOME):
+    on_states = {STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON, STATE_OPEN,
+                 STATE_HOME, STATE_HEAT, STATE_COOL}
+    off_states = {STATE_OFF, STATE_UNLOCKED, STATE_UNKNOWN,
+                  STATE_BELOW_HORIZON, STATE_CLOSED, STATE_NOT_HOME,
+                  STATE_IDLE}
+
+    states = {state.state}
+    if isinstance(state.state, str):
```

Comment:
States are always strings.

Reply:
Thanks. `test_graphite.py` started failing when I assumed they were always strings, since it includes `state = mock.MagicMock(state=0, attributes=attrs)`. Anyway, I've removed this per your request.